frontend/server/src/Controllers/Contest.php

```diff
@@ -1936,21 +1936,26 @@ private static function validateCommonCreateOrUpdate(
         // Show scoreboard is always optional
         $r->ensureOptionalBool('show_scoreboard_after');
 
-        // languages is always optional
-        if (!empty($r['languages'])) {
-            if (is_string($r['languages'])) {
-                $r['languages'] = explode(',', $r['languages']);
-            }
-            if (is_array($r['languages'])) {
-                foreach ($r['languages'] as $language) {
-                    \OmegaUp\Validators::validateOptionalInEnum(
-                        $language,
-                        'languages',
-                        array_keys(
-                            \OmegaUp\Controllers\Run::SUPPORTED_LANGUAGES
-                        )
-                    );
-                }
+        // languages is always required
+        if (empty($r['languages'])) {
```

Comment:
¿Sabes que acabo de notar? Que habrá ocasiones donde requieras editar el concurso y no sea necesario mandar los lenguajes, por ejemplo: cuando un curso cambiará en el modo de acceso. Así que esta condición debería cambiar por:
```suggestion
        // languages is required only when a contest is created
        if ($isRequired && empty($r['languages'])) {
```